src/core/exception.d

```diff
@@ -579,6 +579,12 @@ extern (C) void onSwitchError( string file = __FILE__, size_t line = __LINE__ )
     throw new SwitchError( file, line, null );
 }
 
+// Compiler lowers final switch default case to this (which is a runtime error)
+void __switch_errorT()(string file = __FILE__, size_t line = __LINE__) @trusted
+{
+    // Consider making this a compile time check.
+    throw staticError!SwitchError(file, line, null);
```

Comment:
Keep in mind, that we may need to add a `if (__ctfe)` branch, as I'm not sure if `staticError` would work in CTFE.
I prefer to leave the PR as it is for now, and cross the bridges when we come to them.